Saule/StringExtensions.cs

```diff
@@ -23,7 +23,7 @@ public static string EnsureStartsWith(this string source, string start)
 
         public static string TrimJoin(this char separator, params string[] parts)
         {
-            return string.Join(separator.ToString(), parts.Select(p => p.Trim(separator)));
+            return string.Join(separator.ToString(), parts.Select(p => p?.Trim(separator)));
```

Comment:
In what situation does parts contain `null`? Would we not just want to filter those out instead?